django/test/runner.py

```diff
@@ -114,9 +116,24 @@ def __init__(self):
 
         self.events = []
         self.failfast = False
+        self.buffer = False
         self.shouldStop = False
+        self._stdout_buffer = None
+        self._stderr_buffer = None
+        self._original_stdout = sys.stdout
+        self._original_stderr = sys.stderr
+        self._mirrorOutput = False
```

Comment:
Most of the changes to this class are copied from unittest's `TextResult`: https://github.com/python/cpython/blob/a45b695b9fcfbbb0a087222abc5c8d691a7d2770/Lib/unittest/result.py#L47-L52 . But it was already half copied.

Reply:
> Paging @bmispelon for review since you wrote the first buffer PR #12169.

I don't feel very confident reviewing this. My work on #12169 was mostly to write glue code to pass the option correctly to python's unittest.

One thing I wonder is whether you've investigated actually extending the base `TestResult` instead of copy/pasting from it.
The docstring of `RemoteTestResult` seems to indicate that the copy/pasting was done because Django didn't need the whole API of `TestResult` but I get the feeling that we're trying to implement that API now, so maybe it'd be worth it to extend rather than copy/paste.

Relying on Python's implementation would also make me more confident about the lack of tests.

Reply:
> One thing I wonder is whether you've investigated actually extending the base TestResult instead of copy/pasting from it.

I had not. I guess I preferred to leave the sleeping dog lie, but after further inspection it looks pretty easy. It in fact removes a lot of duplication on our side such as the stop flag and method.

I've "neutered" a method and the list data structures to avoid unnecessarily increasing processing time and memory usage, see comments.